plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -31,12 +34,39 @@
 public class SymbolLookup {
     private final ConcurrentMap<Key,Object> cache = new ConcurrentHashMap<Key, Object>();
 
+    private final ConcurrentMap<Key,Object> noHitCache = new ConcurrentHashMap<Key, Object>();
+
+    static final Object NO_HIT = new Object();
```

Comment:
Overly complex typing here. Since this is merely a flag, it would suffice for this to be a `Map<Key, Boolean>` (or even a `ConcurrentHashSet` if we had one of the libraries providing one handy).

I suppose you are not just folding this into `cache` (retaining `NO_HIT` as a special value) since you expect the code path for cache misses to be less often encountered than positive hits, and wish to avoid calculating the list of plugin names too often.

Reply:
Correct, I also don't want to have to purge both caches.

Reply:
so can simplify to a `Boolean` value type